### ISSUE TITLE
Support partial deserialization

### ISSUE THREAD

**Issue by strowk** [cloudflare/serde-wasm-bindgen#8]
Imagine there's a structure in Js, which I know partly, but some things of it are too dynamic to be mapped to types. 
Let's say in js something would look like this:
``` js
{
    dynamicData: {  someData: "whatever" },
    staticData: { someOtherData: "any" }
}
```

From rust perspective I would like to look at it like this:

``` rust

#[derive(Deserialize)]
struct MyStaticData {
    staticData: String
}

#[derive(Deserialize)]
struct MyData {
    dynamicData: JsValue,
    staticData: MyStaticData,
}

```

So in this example, structure of `MyStaticData` is known beforehand, but for field `dynamicData` we don't know how will it look like at compile time, and so we'd like to just keep there JsValue without changes (and if possible, without copying). 

However this will not compile, because Deserialize trait required for `from_value` cannot be derived for structure with JsValue, because JsValue does not implement it. 

I tried to implement it myself  by wrapping JsValue inside custom RawJsValue  like that :
``` rust
struct RawJsValue {
    value: JsValue
}

impl<'de> Deserialize<'de> for RawJsValue {
...
```

However I'm not sure how exactly to implement Deserialize. I figure that if serde::Deserializer would be of type `serde_wasm_bindgen::Deserializer` , then it's just about rewrapping value, but I don't understand how to cast deserializer to that structure from generic type. Any ideas?

**Comment by RReverser** [comment]
Unfortunately, this isn't possible with Serde. Due to abstraction it uses, the precise types used by specific deserializer - in this case `JsValue` - are lost in translation.

You can deserialize into a `HashMap` or custom `enum` if you know potential shape of your expected value, but you can't preserve `JsValue` itself.

**Comment by RReverser** [comment]
For anyone coming here, this is possible now on Git main (soon to be released) via #40.
